src/magic_folder/snapshot.py

```diff
@@ -144,40 +154,64 @@ def tahoe_create_snapshot_dir(nodeurl, content, parents, timestamp, treq):
 class TahoeSnapshot(object):
     """
     Represents a snapshot corresponding to a file.
+
+    XXX we want a 'file name' of some sort .. is that relative to the
+    magic-folder base, or absolute, or that weird 'flattened' thing
+    magic-folder does?
     """
 
-    filepath = attr.ib()
-    node_directory = attr.ib(converter=lambda p: p.asBytesMode())
+    capability = attr.ib()
+    parents = attr.ib()
 
-    @inlineCallbacks
-    def create_snapshot(self, parents, treq):
-        """
-        Create a snapshot.
 
-        :param [unicode] parents: List of parent snapshots of the current snapshot
-            (read-caps of parent snapshots)
+@inlineCallbacks
+def create_snapshot(node_directory, filepath, parents, treq):
+    """
+    Create a snapshot.
 
-        :param HTTPClient treq: An ``HTTPClient`` or similar object to use to
-            make the queries.
+    :param [unicode] parents: List of parent snapshots of the current snapshot
+        (read-caps of parent snapshots)
```

Comment:
I definitely agree that "unicode string" is the wrong representation .. would have to think more about the rest. A `TahoeSnapshot` should probably know its own capability ... but then I guess the question is: should those only represent "things in the grid", or could/should they also represent "things I might put into the grid sometime but are just in memory right now" (because then the latter kind could have no capability). Probably needs some design and consideration of the implications (and what resulting code looks like).